src/scipp/core/domains.py

```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+from __future__ import annotations
+
+from .._scipp.core import DataArray
+from .shape import concat
+
+
+def find_domains(da: DataArray) -> DataArray:
```

Comment:
Not sure about the name. 'domain' can mean many different things. What about something like `merge_equal_adjacent`?